misk-hibernate-testing/src/main/kotlin/misk/jdbc/VitessScaleSafetyChecks.kt

```diff
@@ -0,0 +1,381 @@
+package misk.jdbc
+
+import com.squareup.moshi.Moshi
+import misk.moshi.adapter
+import misk.okio.split
+import mu.KotlinLogging
+import net.ttddyy.dsproxy.ExecutionInfo
+import net.ttddyy.dsproxy.QueryInfo
+import net.ttddyy.dsproxy.listener.MethodExecutionContext
+import net.ttddyy.dsproxy.listener.MethodExecutionListener
+import net.ttddyy.dsproxy.listener.QueryExecutionListener
+import net.ttddyy.dsproxy.proxy.ProxyConfig
+import net.ttddyy.dsproxy.support.ProxyDataSource
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okio.BufferedSource
+import okio.ByteString.Companion.encodeUtf8
+import java.io.EOFException
+import java.sql.Connection
+import javax.inject.Singleton
+import javax.sql.DataSource
+
+internal data class Instruction(
+  val Opcode: String,
+  val Input: Instruction?
```

Comment:
Uppercase field names are strange . . . Input –> input. Opcode –> opcode

Reply:
It's so they match the json file. I agree it's weird but I really couldn't figure out how to make Moshi use java property names for a json property in Kotlin. Do you know how?

Reply:
Also – probably worth a comment somewhere that announces that this is a JSON thing